### ISSUE TITLE
GeneNet (Java) generated null pointer exception

### ISSUE THREAD

**Issue by zgzn** [MyersResearchGroup/iBioSim#312]
When only one "no influence" arc from Lac I to TetR was added to the attached model, the following null pointer exception happened. This was tested on Red Hat Enterprise Linux, but did not show on the Mac build. With two "no influence" arcs (CI -> TetR and GFP -> TetR), the same error got generated. However, not every combination of two "no influence" arcs generated this exception.

java.lang.NullPointerException
at learn.genenet.SpeciesCollection.getColumn(SpeciesCollection.java:35)
at learn.genenet.Learn.getDiscreteLevels(Learn.java:176)
at learn.genenet.Learn.EncodeExpts(Learn.java:151)
at learn.genenet.Learn.learnNetwork(Learn.java:131)
at learn.genenet.Run.run(Run.java:32)
at learn.GCM.LearnGCM.run(LearnGCM.java:1442)
at java.lang.Thread.run(Thread.java:745)

[Problem3.xml.zip](https://github.com/MyersResearchGroup/iBioSim/files/914209/Problem3.xml.zip)

**Comment by leandrohw** [comment]
I wasn't able to reproduce this. I wonder if there was a particular dataset that caused problems.

**Comment by cjmyers** [comment]
Zhen: any thoughts?

**Comment by zgzn** [comment]
No much. I logged this when it occurred to a student who worked on Red Hat Linux. But I was not able to reproduce it on Mac. @leandrohw What dataset did you refer to?

**Comment by leandrohw** [comment]
I simulated it and used the data to run the learning method. It complains that you need data for promoters that are added when no influence arcs are drew. However, I didn't get the exception.

**Comment by zgzn** [comment]
@leandrohw I assume you tested on Fedora and got no exception?

**Comment by leandrohw** [comment]
@zgzn The error happens because no influence arcs add species to the model and GeneNet tries to load data for those species. Using the model you uploaded didn't cause any problem because I simulated the model so I got values for P0. However, adding a new arc caused problems because there is no time series data for the new promoter. I guess for some reason, the mac version prints an error message and stops before running into the problem.
